File: sample-app/src/main/java/org/granchi/whatnow/WhatNowPreferencesModel.java
package org.granchi.whatnow;

import org.granchi.hollywood.Action;
import org.granchi.hollywood.Model;

import javax.inject.Inject;

/**
 * Model for the app preferences.
 *
 * @author serandel
 */
public class WhatNowPreferencesModel extends Model {
    /**
     * Constructor.
     */
    @Inject
    public WhatNowPreferencesModel() {
        // TODO something
    }

    @Override
    public Model actUpon(Action action) {
        return this;
    }
}

File: sample-app/src/main/kotlin/org/granchi/whatnow/WhatNowPreferencesModel.kt
package org.granchi.whatnow

import org.granchi.hollywood.Action
import org.granchi.hollywood.Model

import javax.inject.Inject

/**
 * Model for the app preferences.
 */
// TODO something
class WhatNowPreferencesModel @Inject constructor() : Model() {
    override fun actUpon(action: Action): Model? {
        return this
    }
}
